File: EclipseProjects/Pull/src/pull/PullingBalancesCalculator.java
package pull;

import java.util.List;

import pull.months.Month;
import pull.months.Months;

import common.BalancesOfMonth;
import common.BalancesOfMonthCalculator;
import common.Transaction;

public class PullingBalancesCalculator implements BalancesOfMonthCalculator
{

	private final List<Transaction> transactions;

	public PullingBalancesCalculator(List<Transaction> transactions)
	{
		super();
		this.transactions = transactions;
	}

	@Override
	public void fillData(List<BalancesOfMonth> balancesOfMonthList)
	{
		Months months = new Months(balancesOfMonthList, transactions);

		for (BalancesOfMonth balancesOfMonth : balancesOfMonthList)
		{
			Month accordingMonth = months.forDate(balancesOfMonth.getDate());
			balancesOfMonth.setBalance(accordingMonth.getBalance());
			balancesOfMonth.setAverageBalance(accordingMonth.getAverageBalance());
		}
	}
}

File: csharp/11_Pull/PullingBalancesCalculator.cs
using System.Collections.Generic;
using Common;
using Pull.Months;

namespace Pull
{
    public class PullingBalancesCalculator : BalancesOfMonthCalculator
    {
        private readonly IList<Transaction> transactions;

        public PullingBalancesCalculator(IList<Transaction> transactions)
        {
            this.transactions = transactions;
        }

        public void FillData(IList<BalancesOfMonth> balancesOfMonthList)
        {
            Months.Months months = new Months.Months(balancesOfMonthList, transactions);

            foreach (BalancesOfMonth balancesOfMonth in balancesOfMonthList)
            {
                ValuesOfMonth valuesOfMonth = months.ForDate(balancesOfMonth.Date);
                balancesOfMonth.Balance = valuesOfMonth.Balance;
                balancesOfMonth.AverageBalance = valuesOfMonth.AverageBalance;
            }
        }
    }
}
